scripts/release-controller.sh

```diff
@@ -49,110 +55,64 @@ check_is_installed git
 eval $ASSUME_COMMAND
-echo "release-controller.sh] [SETUP] Assumed ECR_PUBLISH_ROLE_ARN"
+echo "release-controller.sh] Assumed $ECR_PUBLISH_ROLE_ARN"
 
 # Setup the destination repository for docker and helm
-perform_docker_and_helm_login
+perform_docker_and_helm_login $IMAGE_REPOSITORY
 
-# Do not rebuild controller image for stable releases
-if [[ "$SKIP_IMAGE_BUILD" != "1" ]]; then
-    if ! (echo "$VERSION" | grep -Eq "stable$"); then
-      # Determine parameters for docker-build command
+if [[ "$SKIP_BUILD_IMAGE" != "1" ]]; then
+  # Determine parameters for docker-build command
 
-      cd "$WORKSPACE_DIR"
+  cd "$WORKSPACE_DIR"
 
-      CONTROLLER_GIT_COMMIT=$(git rev-parse HEAD)
-      QUIET=${QUIET:-"false"}
-      BUILD_DATE=$(date +%Y-%m-%dT%H:%M)
-      CONTROLLER_IMAGE_DOCKERFILE_PATH=$WORKSPACE_DIR/Dockerfile
+  CONTROLLER_GIT_COMMIT=$(git rev-parse HEAD)
+  QUIET=${QUIET:-"false"}
+  BUILD_DATE=$(date +%Y-%m-%dT%H:%M)
+  CONTROLLER_IMAGE_DOCKERFILE_PATH=$WORKSPACE_DIR/Dockerfile
 
-      IMG="$IMAGE_REPOSITORY:$IMAGE_TAG"
-      DOCKER_BUILD_CONTEXT="$WORKSPACE_DIR"
+  IMG="$IMAGE_REPOSITORY:$IMAGE_TAG"
+  DOCKER_BUILD_CONTEXT="$WORKSPACE_DIR"
 
-      if [[ $QUIET = "false" ]]; then
-          echo "building aws-gateway-controller image with tag: ${IMG}"
-          echo " git commit: $CONTROLLER_GIT_COMMIT"
-      fi
-
-      # build controller image
-      docker build -t "$IMG" \
-        -f "$CONTROLLER_IMAGE_DOCKERFILE_PATH" \
-        --build-arg service_controller_git_version="$VERSION" \
-        --build-arg service_controller_git_commit="$CONTROLLER_GIT_COMMIT" \
-        --build-arg build_date="$BUILD_DATE" \
-        "${DOCKER_BUILD_CONTEXT}"
-
-      if [ $? -ne 0 ]; then
-        exit 2
-      fi
-
-      echo "Pushing aws-gateway-controller image with tag: ${IMG}"
+  if [[ $QUIET = "false" ]]; then
+      echo "building and pushing aws-gateway-controller image with tag: ${IMG}"
+      echo " git commit: $CONTROLLER_GIT_COMMIT"
+  fi
 
-      docker push "${IMG}"
+  # build controller image
+  if [[ $CI = "true" ]]; then
+    docker buildx build -t "$IMG" \
+      -f "$CONTROLLER_IMAGE_DOCKERFILE_PATH" \
+      --build-arg service_controller_git_version="$VERSION" \
+      --build-arg service_controller_git_commit="$CONTROLLER_GIT_COMMIT" \
+      --build-arg build_date="$BUILD_DATE" \
+      --cache-from type=gha \
+      --cache-to type=gha,mode=max \
+      --push \
+      "${DOCKER_BUILD_CONTEXT}"
+  else
+    docker buildx build -t "$IMG" \
+      -f "$CONTROLLER_IMAGE_DOCKERFILE_PATH" \
+      --build-arg service_controller_git_version="$VERSION" \
+      --build-arg service_controller_git_commit="$CONTROLLER_GIT_COMMIT" \
+      --build-arg build_date="$BUILD_DATE" \
+      --push \
+      "${DOCKER_BUILD_CONTEXT}"
+  fi
```

Comment:
Out of the scope: We can build both amd64 and arm64 image by `--platform` option for the controller public release